juniper-junos/utils.py

```diff
@@ -0,0 +1,172 @@
+    '''
+    address_list = []
+    address_name = []
+    address_set = get_input(params, 'address_set', str)
+    object_type = get_input(params, 'object_type', str)
+    object_to_add = get_input(params, 'object_to_add', str)
+    object_to_delete = get_input(params, 'object_to_delete', str)
+    target_object = object_to_add if object_to_add else object_to_delete
+
+    if ',' in target_object:
+        target_object = target_object.replace(' ', '').split(',')
+        for obj in target_object:
+            if not input_validator(obj,object_type):
+                logger.info('Entered value: {0} is not a valid: {1}'.format(obj,object_type))
+                raise ConnectorError('Entered value: {0} is not a valid: {1}'.format(obj,object_type))
+            if object_to_add:
+                object_definition = ADDRESS_OBJECT[object_type].format(obj)
+                address_list.append(ADDRESS_OBJECT['address-book-add-entry'].format(NAME=obj,ENTRY=object_definition))
+                address_name.append(ADDRESS_OBJECT['address-set-add-entry'].format(NAME=obj))
+            elif object_to_delete:
+                address_list.append(ADDRESS_OBJECT['address-book-delete-entry'].format(NAME=obj))
+                address_name.append(ADDRESS_OBJECT['address-set-delete-entry'].format(NAME=obj))
+    else:
+        if not input_validator(target_object, object_type):
+            logger.info('Entered value: {0} is not a valid: {1}'.format(target_object, object_type))
+            raise ConnectorError('Entered value: {0} is not a valid: {1}'.format(target_object, object_type))
+        if object_to_add:
+            object_definition = ADDRESS_OBJECT[object_type].format(target_object)
+            address_list.append(ADDRESS_OBJECT['address-book-add-entry'].format(NAME=target_object, ENTRY=object_definition))
+            address_name.append(ADDRESS_OBJECT['address-set-add-entry'].format(NAME=target_object))
+        elif object_to_delete:
+            address_list.append(ADDRESS_OBJECT['address-book-delete-entry'].format(NAME=target_object))
+            address_name.append(ADDRESS_OBJECT['address-set-delete-entry'].format(NAME=target_object))
+
+    payload = CONFIG_GLOBAL_ADDRESS_BOOK.format(ADDRESS_LIST=' '.join(address_list),
+                                                ADDRESS_SET_NAME=address_set,
+                                                ADDRESS_NAME_LIST=' '.join(address_name))
+    return payload
+
+def parse_config_xml_response(response):
+    '''
+    Parses JunOS configuration action response and extracts its XML, fixes it and converts it to JSON
+    :param response: Raw server response
+    :return: JSON formatted response
+    '''
+    xml_response = re.findall(r'\<.*\>', response)
+    if len(xml_response) > 0:
+        return_data = '<response>{0}</response>'.format(''.join(xml_response).replace('\\', ''))
+        return xmltodict.parse(return_data)
+
+def parse_op_xml_response(response_data):
```

Comment:
If user passes the output response format as JSON then no need to convert this XML to JSON. The response will come in json. please check this once.